### PR TITLE
Book Delete

### LINKED ISSUE #15: User Story 19, User can delete a book
```
As a Visitor,
When I visit a book's show page,
I see a link on the page to delete the book.
This link should return me to the book index page where I
no longer see this book listed.

(your controller may need to delete other content before you can delete the book)
```

DO THESE STORIES FIRST:
- User Story 10, Book Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #9

child of #33

depends on #20

### PR DESCRIPTION
**Pull request by alexander-mathieu**
Dear Brennan,

This PR add delete functionality from a book's show page.

- Add feature specs covering user experience on /books/show_spec.rb
- Add destroy action to BooksController with confirmation, flash notice and redirect
- Add 'Delete Book' link to view
- Add dependent destroy to Book model for Authors/Reviews

Also included is a minor refactor to the add_authors method in BooksController.

Closes #15 

Warm regards,

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -34,16 +34,21 @@ def create
     end
   end
 
+  def destroy
+    @book = Book.destroy(params[:id])
+
+    flash.notice = "'#{@book.title}' was deleted."
```
Comment (BrennanAyers) [inline app/controllers/books_controller.rb:40]: Nice!

File: app/models/book.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Book < ApplicationRecord
-  has_many :reviews
-  has_many :books_by_author
+  has_many :reviews, dependent: :destroy
+  has_many :books_by_author, dependent: :destroy
```
Comment (BrennanAyers) [inline app/models/book.rb:3]: What does this do?

File: app/views/books/show.html.erb [hunk 3]
```diff
@@ -18,3 +18,5 @@
     </div>
     <% end %>
 </section>
+
+<%= link_to "Delete Book", book_path(@book), method: :delete, data: {confirm: "Are you sure you want to delete the book?"} %>
```
Comment (BrennanAyers) [inline app/views/books/show.html.erb:22]: I like the confirmation too!

File: app/models/book.rb (hunk 2 above)
Comment (alexander-mathieu) [reply app/models/book.rb:3]: It destroys resources that are dependent on the thing being destroyed!

Comment (BrennanAyers) [reply app/models/book.rb:3]: Sweet! Much easier.
